Codecademy/gamut
# feat(Menu)!: updates menu variants and usage guidance

### Overview


Updates Menu variants and usage guidance according to figma. The action variant is now popover, the navigation variant is now fixed and the border has been removed, and the select variant is no more.


### PR Checklist

- [x] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=30420-41119&t=RMGcoWVe4j8kecWm-0
- [x] Related to JIRA ticket: [GM-929]

#### Testing Instructions



In storybook preview confirm you see new usage guidance and variants for Menu.
See monolith and mono PRs for specific testing instructions for Menus used in action.

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/39001)  | [Monolith Env](https://pr-39001-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/8955)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-8955)   |




[GM-929]: https://skillsoftdev.atlassian.net/browse/GM-929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
